util/model_output_parser.py

```diff
@@ -60,6 +61,80 @@ def parse_netcdf(file_id, filename, specification):
     ds.close()
     return var_dfs
 
+def parse_grib(file_id, filename, specification):
+    logger.debug("parse GRIB file {}".format(filename))
+    time_dim = specification["time_dim"]
+    horizontal_dims = specification["horizontal_dims"]
+    fill_value_key = specification.get("fill_value_key", None)
+
+    ds_grib = earthkit.data.from_source("file", filename)
+    short_name_excl = specification["var_excl"]
+
+    short_name = np.unique(ds_grib.metadata("shortName"))
+    short_name = short_name[np.where(~np.isin(short_name, short_name_excl))].tolist()
+
+    level_type = np.unique(ds_grib.metadata("typeOfLevel")).tolist()
+    level_type.remove("unknown")
+
+
+    var_dfs = []
+    for lev in level_type:
+        paramId = np.unique(ds_grib.sel(typeOfLevel=lev, shortName=short_name).metadata("paramId")).tolist()
+        for pid in paramId:
+            ds_temp = get_ds(ds_grib, pid, lev)
+            v = list(ds_temp.keys())[0]
+
+            dim_to_squeeze = [dim for dim, size in zip(ds_temp[v].dims, ds_temp[v].shape) if size==1 and dim != time_dim]
+            ds = ds_temp.squeeze(dim=dim_to_squeeze)
+
+            sub_df = dataframe_from_ncfile(
+                file_id=file_id,
+                filename=filename,
+                varname=v,
+                time_dim=time_dim,
+                horizontal_dims=horizontal_dims,
+                xarray_ds=ds,
+                fill_value_key=fill_value_key,
+            )
+            var_dfs.append(sub_df)
+
+
+    return var_dfs
+
+
+def get_ds(ds_grib, pid, lev):
+    try:
```

Comment:
Not sure I understand this chain of try-except blocks. Is it not possible to do:

`ds = ds_grib.sel(paramId=pid, typeOfLevel=lev, stepType=steps, numberOfPoints=points, stepUnits=unit, dataType=dtype, gridType=gtype).to_xarray()`

Is this because not all parameters have the full spectrum of metadata defined?

Reply:
The GRIB files can only be read when all of those metadata are unique. So we would have to loop through all of them. With the try-except blocks, the files are read faster (by a factor of about 4)

Reply:
So you mean that if you do `ds_grib.sel(paramId=pid, typeOfLevel=lev).to_xarray()` but the selected parameters have, for example, different `stepType`s then an exception is raised?

Reply:
Yes, exactly. But as you mentioned, I have to ensure that not only the last values in the loop are output..

Reply:
👍